### PR TITLE
Add one dotnet6 sample and make Weasel.Core explicit

### PR DESCRIPTION
**Pull request by koide**
I'm suggesting to make the `Weasel.Core` use explicit and proposing a .NET 6 sample. 

If deemed appropriate I could convert the rest of the existing samples to .NET 6 as well.

### CODE REVIEW

**Comment by jeremydmiller** [comment]
@koide Making the enum usage be explicit is fine if that's really that big of a deal. The other sample though, if we're going to use that, at least link to this that makes the clumsy IsDevelopment() syntax unnecessary: https://martendb.io/configuration/optimized_artifact_workflow.html

**Comment by jeremydmiller** [comment]
@koide Sure, but with all due respect, you might be focusing too hard on trivialities IMO.

**Comment by mysticmind** [comment]
> @jeremydmiller I take it that means you don't want this change? Happy to close if so.

@koide quick note, we appreciate and value everyone's input to have a healthy discussion sharing thoughts/perspectives to make Marten better. Please run through my review comments and get back to me with your thoughts.

File: src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+using Marten;
```
[the remark this answers is not in the document]
Comment (mysticmind) [reply src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs:1]: Follow on note, I am just thinking if we even need this sample, probably we can ignore it.

File: src/AspNetCoreWithMarten/Samples/ByStoreOptions/Startup.cs [hunk 2]
```diff
@@ -35,10 +35,12 @@ private StoreOptions BuildStoreOptions()
             options.Connection(connectionString);
 
             // Use the more permissive schema auto create behavior
-            // while in development
```
[the remark this answers is not in the document]
Comment (mysticmind) [reply src/AspNetCoreWithMarten/Samples/ByStoreOptions/Startup.cs:38]: @jeremydmiller please share your thoughts on the above.

File: src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (koide) [reply src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs:1]: Yeah, my intention here was to show how to do it on the new .NET 6 setup, but maybe we can assume that's out of scope for Marten's docs?

File: src/AspNetCoreWithMarten/Samples/ByStoreOptions/Startup.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (koide) [reply src/AspNetCoreWithMarten/Samples/ByStoreOptions/Startup.cs:38]: I just committed a proposal based on your text

File: src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mysticmind) [reply src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs]: At some point, we will update all the docs to the top level/minimal API stuff, until then we can defer adding this. So let us keep it out of scope for now.

[the remark this answers is not in the document]
Comment (mysticmind) [reply src/AspNetCoreWithMarten/Samples/dotnet6/ByNestedClosure/Program.cs]: At some point, we will update all the docs to the top level/minimal API stuff, until then we can defer adding this. So let us keep it out of scope for now.

**Review by mysticmind — APPROVED** [review]
Looks good to me 👍

**Review by oskardudycz — APPROVED** [review]
Looks good to me 👍

**Pull request merged**